### PR TITLE
Support displaying different QR-codes depending on certificate state (EXPOSUREAPP-8704)

### PR DESCRIPTION
**Pull request by d4rken**
If the certificate is `CwaCovidCertificate.State.Invalid` the displayed QR-code will only contain the an [URL to the FAQ](https://www.coronawarn.app/de/faq/#hc_signature_invalid).

* Rename `CwaCovidCertificate.qrCode` to `qrCodeToDisplay`, to reflect that this is not necessarily the certificate QR-code
* Make `qrCodeToDisplay` dependend on `CwaCovidCertificate.State` for Recovery/Vaccination/Test - Certs
* Return `CoilQrCode` instead of `String` so `qrCodeToDisplay` has control over parameters like `errorCorrectionlevel`.
    * Made `CoilQrCode` a `Parcelable` so it can be passed around for navigation.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/storage/RecoveryCertificateContainer.kt [hunk 1]
```diff
@@ -110,8 +112,13 @@ data class RecoveryCertificateContainer(
...
 
-            override val qrCode: QrCodeString
-                get() = data.recoveryCertificateQrCode
+            override val qrCodeToDisplay: CoilQrCode = when (getState()) {
+                State.Invalid -> when (userLocale.language) {
+                    Locale.GERMAN.language -> State.Invalid.URL_INVALID_SIGNATURE_DE
+                    else -> State.Invalid.URL_INVALID_SIGNATURE_EN
+                }.let { CoilQrCode(it, QrCodeOptions(correctionLevel = ErrorCorrectionLevel.M)) }
+                else -> CoilQrCode(data.recoveryCertificateQrCode)
+            }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/storage/RecoveryCertificateContainer.kt:121]: Magic happens here :mage_woman:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationContainer.kt [hunk 2]
```diff
@@ -134,8 +137,13 @@ data class VaccinationContainer internal constructor(
...
 
-        override val qrCode: QrCodeString
-            get() = vaccinationQrCode
+        override val qrCodeToDisplay: CoilQrCode = when (getState()) {
+            State.Invalid -> when (userLocale.language) {
+                Locale.GERMAN.language -> State.Invalid.URL_INVALID_SIGNATURE_DE
+                else -> State.Invalid.URL_INVALID_SIGNATURE_EN
+            }.let { CoilQrCode(it, QrCodeOptions(correctionLevel = ErrorCorrectionLevel.M)) }
+            else -> CoilQrCode(vaccinationQrCode)
+        }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationContainer.kt:146]: Magic happens here :mage_woman:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/storage/TestCertificateContainer.kt [hunk 3]
```diff
@@ -132,8 +134,13 @@ data class TestCertificateContainer(
...
 
-            override val qrCode: QrCodeString
-                get() = data.testCertificateQrCode!!
+            override val qrCodeToDisplay: CoilQrCode = when (getState()) {
+                State.Invalid -> when (userLocale.language) {
+                    Locale.GERMAN.language -> State.Invalid.URL_INVALID_SIGNATURE_DE
+                    else -> State.Invalid.URL_INVALID_SIGNATURE_EN
+                }.let { CoilQrCode(it, QrCodeOptions(correctionLevel = ErrorCorrectionLevel.M)) }
+                else -> CoilQrCode(data.testCertificateQrCode!!)
+            }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/storage/TestCertificateContainer.kt:143]: Magic happens here :mage_woman:

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/storage/TestCertificateContainer.kt:143]: can we move the magic snippet into a common magic place ?  😉

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/repository/CertificateRepoContainer.kt [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+        state: CwaCovidCertificate.State,
+        language: String,
+        certificateQrCode: String
+    ) = when (state) {
+        CwaCovidCertificate.State.Invalid -> when (language) {
+            Locale.GERMAN.language -> CwaCovidCertificate.State.Invalid.URL_INVALID_SIGNATURE_DE
+            else -> CwaCovidCertificate.State.Invalid.URL_INVALID_SIGNATURE_EN
+        }.let { CoilQrCode(it, QrCodeOptions(correctionLevel = ErrorCorrectionLevel.M)) }
+        else -> CoilQrCode(certificateQrCode)
+    }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/repository/CertificateRepoContainer.kt:29]: https://github.com/corona-warn-app/cwa-app-android/pull/3814#discussion_r677528529

**Review by mtwalli — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Pull request merged**